xblock/fields.py

```diff
@@ -779,6 +779,18 @@ def from_json(self, value):
 
     enforce_type = from_json
 
+    def to_string(self, value):
+        """
+        In python3, json.dumps() cannot sort keys of different types, 
+        so preconvert None to 'null'.
+        """
+        self.enforce_type(value)
+        if isinstance(value, dict) and None in value:
+            value = value.copy()
+            value['null'] = value[None]
+            del value[None]
+        return super(Dict, self).to_string(value)
```

Comment:
I hadn't seen that.  The pushback against fixing it is not surprising, but it is frustrating.  You shouldn't be able to break json.dumps just by specifying a cosmetic option. :(

One easyish option that I've played with is to leave in the sorted version, but catch the TypeError, and retry with sort_keys=False.